### PR TITLE
move taint keys to k8s.io/cloud-provider

### PR DESCRIPTION
**Pull request by yastij**
Signed-off-by: Yassine TIJANI <ytijani@vmware.com>

**What type of PR is this?**
/kind cleanup
/area code-organization
/priority important-soon

**What this PR does / why we need it**: This PR moves taint keys to `k8s.io/cloud-provider`. these taints are referenced mostly under cloud control loops.

**Which issue(s) this PR fixes**: Part of https://github.com/kubernetes/kubernetes/issues/81172

**Special notes for your reviewer**:

/assign @andrewsykim @liggitt 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/api/well_known_labels.go [hunk 1]
```diff
@@ -57,15 +57,6 @@ const (
 	// and removed when node has enough disk.
 	TaintNodePIDPressure = "node.kubernetes.io/pid-pressure"
 
-	// TaintExternalCloudProvider sets this taint on a node to mark it as unusable,
-	// when kubelet is started with the "external" cloud provider, until a controller
-	// from the cloud-controller-manager intitializes this node, and then removes
-	// the taint
-	TaintExternalCloudProvider = "node.cloudprovider.kubernetes.io/uninitialized"
```
Comment (andrewsykim) [inline pkg/scheduler/api/well_known_labels.go]: I can't recall the issue, but wasn't there some ongoing work to move schedulerapi somewhere else that's more easily consumable? Maybe @ahg-g knows?

Comment (yastij) [reply pkg/scheduler/api/well_known_labels.go]: https://github.com/kubernetes/kubernetes/issues/82390 is the issue. these taints are only used by the cloud-provider control loops, so it seems natural to move these here

Comment (andrewsykim) [reply pkg/scheduler/api/well_known_labels.go]: What I like about this location is that the taints are considered APIs, in the new location it seems easier for someone to change them without understanding the implications. Wondering if we should have a `k8s.io/cloud-provider/api` package or something? We have other constants that might be a better fit there -- see [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cloud-provider/service/helpers/helper.go#L33-L38) and [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cloud-provider/volume/constants.go#L19-L26). Thoughts @cheftako?

Comment (yastij) [reply pkg/scheduler/api/well_known_labels.go]: I'm fine with `/api`

Comment (ahg-g) [reply pkg/scheduler/api/well_known_labels.go]: here is the api cleanup issue: https://github.com/kubernetes/kubernetes/pull/83578, well_known_labels.go seems to be the only file that will stay under scheduler/api, @Huang-Wei what is the plan for this file?

Comment (liggitt) [reply pkg/scheduler/api/well_known_labels.go]: > Wondering if we should have a `k8s.io/cloud-provider/api` package or something? We have other constants that might be a better fit there -- see [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cloud-provider/service/helpers/helper.go#L33-L38) and [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cloud-provider/volume/constants.go#L19-L26). Thoughts @cheftako?

seems reasonable to me

Comment (Huang-Wei) [reply pkg/scheduler/api/well_known_labels.go]: @ahg-g Here is a list which refers `k8s.io/kubernetes/pkg/scheduler/api`:

```
plugin/pkg/admission/defaulttolerationseconds/admission.go
28:	schedulerapi "k8s.io/kubernetes/pkg/scheduler/api"

plugin/pkg/admission/podtolerationrestriction/admission.go
38:	schedulerapi "k8s.io/kubernetes/pkg/scheduler/api"

pkg/scheduler/algorithm/predicates/predicates.go
46:	legacyapi "k8s.io/kubernetes/pkg/scheduler/api"

pkg/scheduler/algorithm/types.go
26:	legacyapi "k8s.io/kubernetes/pkg/scheduler/api"

pkg/scheduler/api/BUILD
11:    importpath = "k8s.io/kubernetes/pkg/scheduler/api",

pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go
45:	schedulerapi "k8s.io/kubernetes/pkg/scheduler/api"

pkg/controller/daemon/util/daemonset_util.go
29:	schedulerapi "k8s.io/kubernetes/pkg/scheduler/api"

pkg/controller/cloud/node_lifecycle_controller.go
40:	schedulerapi "k8s.io/kubernetes/pkg/scheduler/api"

pkg/controller/cloud/node_controller.go
41:	schedulerapi "k8s.io/kubernetes/pkg/scheduler/api"

pkg/controller/nodelifecycle/node_lifecycle_controller.go
63:	schedulerapi "k8s.io/kubernetes/pkg/scheduler/api"

pkg/kubelet/kubelet_node_status.go
44:	schedulerapi "k8s.io/kubernetes/pkg/scheduler/api"

pkg/kubelet/eviction/eviction_manager.go
43:	schedulerapi "k8s.io/kubernetes/pkg/scheduler/api"
```

It seems the usage are not quite related with cloud-provider, so I'd suggest to move the const strings in `pkg/scheduler/api/well_known_labels.go` to a more common place, such as `staging/src/k8s.io/api/core/v1/well_known_labels.go`. WDYT?

**Comment by ahg-g** [comment]
Hi @yastij what is the status of this PR? asking because we would like to cleanup the ```pkg/scheduler/api``` package.

**Comment by Huang-Wei** [comment]
@yastij Any luck on pushing this PR forward?

**Review by andrewsykim — COMMENTED** [review]
/close

Dupe of https://github.com/kubernetes/kubernetes/pull/88435

**Pull request closed without merging**